java/tools/src/main/java/org/apache/arrow/tools/EchoServer.java

```diff
@@ -23,8 +23,8 @@
 import org.apache.arrow.memory.BufferAllocator;
 import org.apache.arrow.memory.RootAllocator;
 import org.apache.arrow.vector.VectorSchemaRoot;
-import org.apache.arrow.vector.stream.ArrowStreamReader;
-import org.apache.arrow.vector.stream.ArrowStreamWriter;
+import org.apache.arrow.vector.ipc.stream.ArrowStreamReader;
+import org.apache.arrow.vector.ipc.stream.ArrowStreamWriter;
```

Comment:
I would either make this `ipc.ArrowStreamReader` but not `ipc.stream`

Reply:
Do you think the same for file and json readers, e.g. `ipc.ArrowFileReader`?  I most made these subpackages because there were some supporting files specific to just the file reader, so they could be grouped together.  But I'm ok either way, @icexelloss brought this up here https://github.com/apache/arrow/pull/1259#issuecomment-340562836

Reply:
These classes are all quite similar (the file format is very nearly the stream format, plus a file footer and magic numbers at start and end), I think it would make sense to keep them in a flat package namespace (but I'm not a Java expert)

Reply:
@icexelloss do you have an opinion on this? Would be good to get this patch in soon to facilitate testing

Reply:
I prefer `ipc.ArrowStreamReader` to `ipc.stream.ArrowStreamReader`

Reply:
Sure, I'm fine with this.  I'll change it now